core/src/avm2/object/namespace_object.rs

```diff
@@ -84,8 +81,8 @@ impl<'gc> NamespaceObject<'gc> {
             NamespaceObjectData {
                 base,
                 namespace: Lock::new(namespace),
-                prefix: Lock::new(if namespace.as_uri().is_empty() {
-                    Some("".into())
+                prefix: Lock::new(if namespace.as_uri(activation.strings()).is_empty() {
+                    Some(activation.strings().empty())
```

Comment:
This seems a bit redundant, but I don't see a better solution

Reply:
We could have a dedicated method `namespace.has_empty_uri()`, but I'm not sure this is worth it.